src/pages/index.js

```diff
@@ -22,96 +34,205 @@ const profileDescriptionInput = profileEditModal.querySelector(
   "#profile-description-input"
 );
 
-//Add New Card
-const addCardModal = document.querySelector("#add-card-modal");
+// Edit Avatar
+const avatarEditBtn = document.querySelector("#avatar-edit-button");
+const avatarForm = document.forms["modal__form_avatar"];
+
+const profileImage = document.querySelector(".profile__image");
+
+// Add New Card
 const addCardForm = document.forms["card-form"];
 const addCardBtn = document.querySelector(".profile__add-button");
 
-const renderCard = (cardData) => {
-  const card = createCard(cardData);
-  section.addItem(card);
-};
+// UserInfo
+const user = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
 /* -------------------------------------------------------------------------- */
 /*                                  Popups                                    */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Popup
+
+// Edit Profile Popup
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  (profileData) => {
+    editProfilePopup.renderLoading(true);
+
+    api
+      .editProfile(profileData.title, profileData.subheader)
+      .then((updatedUserInfo) => {
+        user.setUserInfo(updatedUserInfo.name, updatedUserInfo.about);
+        editProfilePopup.close();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => {
+        editProfilePopup.renderLoading(false);
+      });
+  }
 );
 editProfilePopup.setEventListeners();
 
-//Add New Card Popup
-const newCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+// Add New Card Popup
+const newCardPopup = new PopupWithForm("#add-card-modal", (newCardData) => {
+  newCardPopup.renderLoading(true);
+
+  api
+    .addNewCard({ name: newCardData.title, link: newCardData.url })
+    .then((cardData) => {
+      renderCard(cardData);
+      newCardPopup.close();
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      newCardPopup.renderLoading(false);
+    });
+});
 newCardPopup.setEventListeners();
 
-//Preview Image Popup
+// Preview Image Popup
 const previewImagePopup = new PopupWithImage("#preview-modal");
 previewImagePopup.setEventListeners();
 
-//Section
+// Section to render cards
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: renderCard,
   },
   ".cards__list"
 );
-section.renderItems();
 
-//UserInfo
-const user = new UserInfo(".profile__title", ".profile__description");
+// Edit Avatar Popup
+const avatarEditPopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  (formData) => {
+    avatarEditPopup.renderLoading(true);
+    avatarFormValidator.toggleButtonState();
```

Comment:
`avatarFormValidator.toggleButtonState()` is not needed in line 113. You have it on opening